cmd/ww/server.go

```diff
@@ -338,13 +373,22 @@ func server(args ...string) {
 		Prompt:     autocert.AcceptTOS,
 		HostPolicy: autocert.HostWhitelist(strings.Split(*whitelist, ",")...),
 	}
+
+	if customCertFlag != "" {
+		log.Println("Using local certificate and key")
+		CustomGetCertificate = PickLocalCert
+	} else {
+		log.Println("Generating acme certificate")
+		CustomGetCertificate = m.GetCertificate
+	}
+
 	ssrv := &http.Server{
 		ReadTimeout:  10 * time.Second,
 		WriteTimeout: 60 * time.Minute,
 		IdleTimeout:  20 * time.Second,
 		Addr:         *httpsaddr,
 		Handler:      http.HandlerFunc(handler),
-		TLSConfig:    &tls.Config{GetCertificate: m.GetCertificate},
+		TLSConfig:    &tls.Config{GetCertificate: CustomGetCertificate},
```

Comment:
how about we set GetCertificate to nil if we have a custom cert, and always pass the cert/key args to ssrv.ListenAndServeTLS?

this way we get an error if there's something wrong with the files on start, rather than first request. as a bonus, the two args no longer have to be global.

Reply:
Should be ok now, please check if the commit I've just pushed is what you have in mind...